misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -209,7 +210,15 @@ internal class RealTransacter private constructor(
     }
 
     override fun <T : DbEntity<T>> load(id: Id<T>, type: KClass<T>): T {
-      return session.get(type.java, id)
+      try {
+        return session.get(type.java, id)
+      } catch (e : TypeMismatchException) {
```

Comment:
Instead of catching and rethrowing can you instead check for the specific error as a precondition and throw the error there. I suspect `TypeMismatchException` could also be caused by other errors which aren't explained by this specific exception.

Reply:
I'm a bit unclear on what you're suggesting here:
```check for specific error as a precondition, and throw the error there```

This exception is effectively a subset of `TypeMismatchException`, so it sounds like what you want to do is check that we have a `InvalidChildEntityException` before even making the `session.get` call, so that we don't falsely group a `TypeMismatch` as a `InvalidChildEntity`. Does this sound about right?

Reply:
Correct.